.github/workflows/ci.yaml

```diff
@@ -57,16 +59,48 @@ jobs:
         run: |
           composer update --no-interaction --prefer-dist --no-progress ${{ matrix.dependencies }}
 
-      - name: Run tests using Makefile
-        if: runner.os != 'Windows'
+      - name: Run tests
+        shell: bash
         run: |
           make test-unit
 
-      - name: Run tests without Makefile
+      - name: Run integration tests
+        shell: bash
+        run: |
+          make test-unit PHPUNIT_GROUP=integration
+
+      - name: Cache E2E tests dependencies
         if: runner.os == 'Windows'
+        uses: actions/cache@v2
+        with:
+          path: tests/e2e/*/vendor
+          key: e2e-vendor-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('tests/e2e/*/composer.json') }}-${{ matrix.dependencies }}
+          restore-keys: |
+            e2e-vendor-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('tests/e2e/*/composer.json') }}-
+            e2e-vendor-${{ runner.os }}-${{ matrix.php-version }}-
+            e2e-vendor-${{ runner.os }}-
+            composer-
+
+      - name: Install Composer dependencies for E2E tests
+        shell: bash
         run: |
-          vendor/bin/phpunit --group default
+          ls tests/e2e/*/composer.json | xargs dirname |
+             xargs -I{} composer --working-dir={} update --no-interaction --prefer-dist --no-progress ${{ matrix.dependencies }}
 
-      - name: Run Infection
+      - name: Run a subset of E2E tests
+        if: runner.os == 'Windows'
+        shell: bash
+        run: |
+          make test-e2e-phpunit PHP='phpdbg -qrr' E2E_PHPUNIT_GROUP=e2e BENCHMARK_SOURCES=
```

Comment:
It just wouldn't work with Xdebug. [It seems installed, but then it is not.](https://github.com/infection/infection/pull/1365/checks?check_run_id=1297037605#step:11:1)